### Python Review Assignment
16 points

Due by midnight January 30, 2024


**To avoid erasing the sample output, please insert a new cell above the sample output**. Write your script inside the new cell and run. Verify your output with the sample output

For lab submission, please rename your file using the following format

`firstname_familyname_python_review.ipynb`

In [186]:
# please enter your name and 700# in this cell
# Akhilandeswari Vegi and 700758173
print("Akhilandeswari Vegi")
print(700758173)

Akhilandeswari Vegi
700758173


### Task 1
Define a string `s1` with content 'Good morning John'. Count the number of characters in `s1`

In [386]:
s1 = 'Good morning John'
print(len(s1))


17


### Task 2
Check if the word 'morning' is contained in `s1`

In [387]:
s1= 'Good morning John'

In [388]:
if 'morning' in s1:
    print("True")

True


### Task 3
Use a function to replace 'John' with 'David' in `s1`. Show the content of the new string

In [389]:
s1='Good morning John'

In [390]:
new_string = s1.replace("John","David")
print(new_string)

Good morning David


### Task 4
Use your UCM email address to create a string s2. Extract the string `ucmo.edu`

In [391]:
s2 = 'axv81730@ucmo.edu'

In [392]:
print(s2[-8:]) # characters from the eight-last (included) to the end

ucmo.edu


### Task 5
Create a list x with integers from 1 to 30. Create a new list y which only conatins the odd numbers from list x. You are not allowed to mannually type out the lists

In [393]:

x=[] # input list to store integers from 1 to 30 without entering manually
y=[] # output list to store odd numbers from x
for i in range(1,31):
    x.append(i)

for i in x:
    if i%2 == 1:
        y.append(i)
print(y) 


[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29]


### Task 6
Create a new list z with each element is a squre of the corresponding element from list y. You are not allowed to mannually type out the list

In [394]:
z=[]
for i in y:
    z.append(i**2)
print(z)

[1, 9, 25, 49, 81, 121, 169, 225, 289, 361, 441, 529, 625, 729, 841]


### Task 7
Extract `[729, 625, 529, 441, 361, 289, 225, 169]` from list z. You are not allowed to mannually type out the list

In [395]:
reversed_list = []

for i in range(len(z)-2,5,-1):
    reversed_list.append(z[i])

print(reversed_list)


[729, 625, 529, 441, 361, 289, 225, 169]


### Task 8
Reverse the order of z. You are not allowed to mannually type out the list

In [396]:
reversedList = z
print(reversedList[::-1])

[841, 729, 625, 529, 441, 361, 289, 225, 169, 121, 81, 49, 25, 9, 1]


### Task 9
Given the following list, extract the string 'Get Me'

In [397]:
lst = [1, 2, [3, 4], [5, [100, 200, ['Get Me']], 300, 11], 1, 9]

In [398]:
print(lst[-3][-3][-1][0])

Get Me


### Task 10
Sort the following list of strings by their lengths. You are not allowed to mannually type out the list

In [399]:
x = ['ccc', 'dddd', 'bb', 'ggggggg', 'a', 'eeeee', 'ffffff']

In [400]:
def SortList(x):
    y = sorted(x, key=len)
    return y
print(SortList(x))

['a', 'bb', 'ccc', 'dddd', 'eeeee', 'ffffff', 'ggggggg']


### Task 11
Extract the string 'Get Me' from the following dictionary d

In [401]:
d = {'k1':[1, 2, 3, {'tricky':['oh', 'man', 'crazy', {'target':[1, 2, 3, 'Get Me']}]}]}

In [402]:
target_val = d['k1'][3]['tricky'][3]['target'][3]
print(target_val)

Get Me


### Task 12
Use lambda expressions and the `filter()` function to filter out words from the following list that don't start with the letter 's'

In [403]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']


In [404]:
output_seq = filter(lambda word: word[0] !='s', seq)

print(list(output_seq))

['apple', 'banana', 'kiwi']


### Task 13
Count the number of unique characters in the following string s. Please ignore the white space and character case. For example, 'P' and 'p' are considered as one character

In [405]:
s = 'Python programming is fun'


In [406]:
s = s.lower().replace(" ", "")

unique_characterString = set(s) # Convert to a set

count = len(unique_characterString)

print(count)

14


### Task 14
Use lambda expressions and the `map()` function to count the length of each string in the list

In [407]:
seq = ['soup', 'apple', 'salad', 'banana', 'spaghetti', 'spinach', 'kiwi']


In [408]:
result_seq = list(map(lambda x: len(x), seq))
print(result_seq)

[4, 5, 5, 6, 9, 7, 4]


### Task 15
Create a function that counts the number of times the word 'dog' occurs in a string. Ignore edge cases. 

In [409]:
def countDog(string):
    return string.count('dog')


In [410]:
countDog('This dog looks differently from that dog in the dog shelter!')

3

### Task 16
Write a `for` loop to process employee information listed in the cell below. For each employee with Level 1, increase the salary by 2000 and adjust the Level from 1 to 2. Display all employee information afterwards.

In [411]:
employee = {
    'Elon Musk':{'Department':'Sales', 'Salary':6000, 'Level':2},
    'Bill Gates': {'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Donald Trump':{'Department':'Public Relations', 'Salary':3000, 'Level':1},
    'Joe Biden':{'Department':'Human Resources', 'Salary':5000, 'Level':2},
    'Mark Zuckerberg':{'Department':'Engineering', 'Salary':10000, 'Level':3},
    'Jensen Huang':{'Department':'Sales', 'Salary':4000, 'Level':1}
}

In [412]:
for key in employee:
    if employee[key]['Level'] == 1:
        employee[key]['Salary'] += 2000
        employee[key]['Level'] = 2

print(employee)

{'Elon Musk': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}, 'Bill Gates': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Donald Trump': {'Department': 'Public Relations', 'Salary': 5000, 'Level': 2}, 'Joe Biden': {'Department': 'Human Resources', 'Salary': 5000, 'Level': 2}, 'Mark Zuckerberg': {'Department': 'Engineering', 'Salary': 10000, 'Level': 3}, 'Jensen Huang': {'Department': 'Sales', 'Salary': 6000, 'Level': 2}}
